test/files/stacktrace.json

```diff
@@ -0,0 +1 @@
cation.AppMainV2","method":"main","file":"AppMainV2.java","line":131,"exact":true,"location":"idea_rt.jar","version":"?"},{"class":"App","method":"main","file":"App.java","line":15,"exact":true,"location":"classes/","version":"?"},{"class":"jdk.internal.reflect.NativeMethodAccessorImpl","method":"invoke0","file":"NativeMethodAccessorImpl.java","line":-2,"exact":false,"location":"?","version":"?"},{"class":"jdk.internal.reflect.NativeMethodAccessorImpl","method":"invoke","file":"NativeMethodAccessorImpl.java","line":62,"exact":false,"location":"?","version":"?"},{"class":"jdk.internal.reflect.DelegatingMethodAccessorImpl","method":"invoke","file":"DelegatingMethodAccessorImpl.java","line":43,"exact":false,"location":"?","version":"?"},{"class":"java.lang.reflect.Method","method":"invoke","file":"Method.java","line":566,"exact":false,"location":"?","version":"?"},{"class":"com.intellij.rt.execution.application.AppMainV2","method":"main","file":"AppMainV2.java","line":131,"exact":true,"location":"idea_rt.jar","version":"?"},{"class":"App","method":"main","file":"App.java","line":15,"exact":true,"location":"classes/","version":"?"},{"class":"jdk.internal.reflect.NativeMethodAccessorImpl","method":"invoke0","file":"NativeMethodAccessorImpl.java","line":-2,"exact":false,"location":"?","version":"?"},{"class":"jdk.internal.reflect.NativeMethodAccessorImpl","method":"invoke","file":"NativeMethodAccessorImpl.java","line":62,"exact":false,"location":"?","version":"?"},{"class":"jdk.internal.reflect.DelegatingMethodAccessorImpl","method":"invoke","file":"DelegatingMethodAccessorImpl.java","line":43,"exact":false,"location":"?","version":"?"},{"class":"java.lang.reflect.Method","method":"invoke","file":"Method.java","line":566,"exact":false,"location":"?","version":"?"},{"class":"com.intellij.rt.execution.application.AppMainV2","method":"main","file":"AppMainV2.java","line":131,"exact":true,"location":"idea_rt.jar","version":"?"},{"class":"App","method":"main","file":"App.java","line":15,"exact":true,"location":"classes/","version":"?"},{"class":"jdk.internal.reflect.NativeMethodAccessorImpl","method":"invoke0","file":"NativeMethodAccessorImpl.java","line":-2,"exact":false,"location":"?","version":"?"},{"class":"jdk.internal.reflect.NativeMethodAccessorImpl","method":"invoke","file":"NativeMethodAccessorImpl.java","line":62,"exact":false,"location":"?","version":"?"},{"class":"jdk.internal.reflect.DelegatingMethodAccessorImpl","method":"invoke","file":"DelegatingMethodAccessorImpl.java","line":43,"exact":false,"location":"?","version":"?"},{"class":"java.lang.reflect.Method","method":"invoke","file":"Method.java","line":566,"exact":false,"location":"?","version":"?"},{"class":"com.intellij.rt.execution.application.AppMainV2","method":"main","file":"AppMainV2.java","line":131,"exact":true,"location":"idea_rt.jar","version":"?"}]},"endOfBatch":false,"loggerFqcn":"org.apache.logging.log4j.spi.AbstractLogger","threadId":1,"threadPriority":5,"service":"java-app"}
```

Comment:
Do we have the log formatter details (e.g. log4j formatter) so we can document what clients may need to use.  My expectation would have been a single JSON where something like the message field encapsulated the entire stack trace instead of individual entries for each line of the stack trace.

Reply:
Yes this is log4j format:
```
<?xml version="1.0" encoding="UTF-8"?>
<Configuration status="WARN">
  <Appenders>
    <Console name="console" target="SYSTEM_OUT">
      <JsonLayout compact="true" eventEol="true" >
        <KeyValuePair key="service" value="java-app" />
      </JsonLayout>
    </Console>
  </Appenders>
  <Loggers>
    <Root level="info">
      <AppenderRef ref="console" />
    </Root>
  </Loggers>
</Configuration>
```

Reply:
any idea if this is a recommended or common formatter?

Reply:
I would say it's recommended settings for getting one record per line.

- `compact="true"` - If true, the appender does not use end-of-lines and indentation. Defaults to false.
- `eventEol="true"` - If true, the appender appends an end-of-line after each record. Defaults to false. Use with eventEol=true and compact=true to get one record per line.

More here: https://logging.apache.org/log4j/2.x/manual/layouts.html#JSONLayout

Reply:
@sichvoge   this is info for configuring java apps log formatter.  Thoughts if we should make some recommendations in our documentation?